src/network/core.rs

```diff
@@ -0,0 +1,128 @@
+use crate::network::{core_utils, types};
+use nix::sched;
+use rand::Rng;
+use std::fs::File;
+use std::io::Error;
+use std::os::unix::prelude::*;
+use std::thread;
+
+pub struct Core {
+    pub networkns: String,
+}
+
+impl Core {
+    pub fn bridge_per_podman_network(
+        network_opts: &types::NetworkOptions,
+        _netns: &str,
+    ) -> Result<(), std::io::Error> {
+        for (net_name, network) in &network_opts.network_info {
+            let network_name: String = net_name.to_owned();
+            // get PerNetworkOptions for this network
+            let network_per_opts = network_opts.networks.get(&network_name);
+            // get bridge name
+            let _br_name: String = network.network_interface.as_ref().unwrap().to_owned();
+            let _br_subnets = &network.subnets;
+
+            //we have the bridge name but we must iterate for all the available gateways
+            for subnet in network.subnets.as_ref().unwrap() {
+                let _gw_ipaddr: String = subnet.gateway.as_ref().unwrap().to_owned();
+                let _mask = subnet.subnet.to_string();
+                let _container_veth_name: String =
+                    network_per_opts.unwrap().interface_name.to_owned();
+                let static_ips: &Vec<String> =
+                    network_per_opts.unwrap().static_ips.as_ref().unwrap();
+                for ip in static_ips {
+                    let _ip_addr: String = ip.to_owned();
+                    //call
+                    //add_bridge_and_veth(&br_name, ip_addr, mask, gw_ipaddr, &container_veth_name,
+                    //netns);
+                }
+            }
+        }
+        Ok(())
+    }
+
+    pub fn add_bridge_and_veth(
+        br_name: &str,
+        netns_ipaddr: &str,
+        netns_ipaddr_mask: &str,
+        gw_ipaddr: &str,
```

Comment:
Note that we have to support n ips/subnets per bridge. Right now you only seem to configure one.